src/main/java/com/github/ermay12/regex/Regex.java

```diff
@@ -891,9 +939,17 @@ public String replace(CharSequence input, ReplacementRegex replacement) {
 
   public String replace(CharSequence input, Function<RegexMatch, String> l) {
     Matcher m = getMatcher(input);
-    return m.replaceAll(match -> l.apply(new RegexMatch(match, this)));
+    StringBuilder sb = new StringBuilder();
```

Comment:
Shouldnt you be able to do this by using the stream that we have?

Reply:
Something about needing the matcher and not just a match result i think

Reply:
fine